axum-extra/Cargo.toml

```diff
@@ -14,16 +14,23 @@ version = "0.3.5"
 [features]
 default = []
 
-async-read-body = ["tokio-util/io"]
-cookie = ["cookie-lib"]
-cookie-private = ["cookie", "cookie-lib/private"]
-cookie-signed = ["cookie", "cookie-lib/signed"]
-erased-json = ["serde_json", "serde"]
-form = ["serde", "serde_html_form"]
-json-lines = ["serde_json", "serde", "tokio-util/io", "tokio-stream/io-util"]
-query = ["serde", "serde_html_form"]
+async-read-body = ["dep:tokio-util", "tokio-util?/io"]
```

Comment:
What does the `?` do?

Reply:
It enables that feature on the dependency only if that dependency is activated by something else. It's also necessary to refer to features of optional dependencies that are activated using `dep:` syntax (otherwise it would be unnecessary here).

Reply:
Ah I see!